include/dr/mhp/algorithms/inclusive_exclusive_scan_impl.hpp

```diff
@@ -110,24 +139,27 @@ auto inclusive_scan_impl_(R &&r, O &&d_first, BinaryOp &&binary_op,
         auto local_in = dr::ranges::__detail::local(global_in);
         auto local_out = rng::views::take(
             dr::ranges::__detail::local(global_out), rng::size(local_in));
-        // dr::drlog.debug("rebase before: {}\n", local_out);
+        auto local_out_begin_direct =
+            detail::direct_iterator(local_out.begin());
+        *local_out_begin_direct = offset;
+        auto local_out_drop = local_out | rng::views::drop(1);
+        // dr::drlog.debug("rebase before: {}\n", local_out_drop);
         if (use_sycl) {
 #ifdef SYCL_LANGUAGE_VERSION
-          auto wrap_rebase = [rebase, base = rng::begin(local_out)](auto idx) {
-            rebase(base[idx]);
-          };
+          auto wrap_rebase = [rebase, base = rng::begin(local_out_drop)](
+                                 auto idx) { rebase(base[idx]); };
           detail::parallel_for(dr::mhp::sycl_queue(),
-                               sycl::range<>(rng::distance(local_out)),
+                               sycl::range<>(rng::distance(local_out_drop)),
```

Comment:
This change is effective for both inclusive and excluisve scan. I think would be faster for inclusive scan if in case of incusive scan old behavior is preserved (no drop, rebase called on whole range)